components/cd-button/cd-button.html

```diff
@@ -21,49 +20,117 @@
 
       <p></p>
 
-      <button type="button" class="cd-button cd-button--style cd-button--small">
+      <button type="button" class="cd-button cd-button--outline cd-button--icon">
+        <svg class="cd-icon cd-icon--user" aria-hidden="true" focusable="false" width="16" height="16">
+          <use xlink:href="#cd-icon--user"></use>
+        </svg>
         <span class="cd-button__text">Button C</span>
       </button>
 
       <p></p>
 
-      <button type="button" class="cd-button cd-button--style">
+      <button type="button" class="cd-button cd-button--style cd-button--wide">
         <span class="cd-button__text">Button D</span>
       </button>
 
       <p></p>
 
-      <button type="button" class="cd-button cd-button--style cd-button--bold cd-button--uppercase">
+      <button type="button" class="cd-button cd-button--bold cd-button--uppercase">
         <span class="cd-button__text">Button E</span>
       </button>
 
       <p></p>
 
-      <a href="#" type="button" class="cd-button cd-button--bold">
+      <a href="#" class="cd-button cd-button--bold">
         Anchor styled as a button
       </a>
 
       <p></p>
 
-      <button type="button" class="cd-button cd-button--style cd-button--icon">
-        <svg class="cd-icon cd-icon--user">
+      <a href="#" class="cd-button cd-button--bold cd-button--icon">
+        <svg class="cd-icon cd-icon--user" aria-hidden="true" focusable="false" width="16" height="16">
+          <use xlink:href="#cd-icon--user"></use>
+        </svg>
+        <span class="cd-button__text">Anchor with icon</span>
+      </a>
+
+      <p></p>
+
+      <button type="button" class="cd-button cd-button--icon">
+        <svg class="cd-icon cd-icon--user" aria-hidden="true" focusable="false" width="16" height="16">
           <use xlink:href="#cd-icon--user"></use>
         </svg>
         <span class="cd-button__text">Button F</span>
       </button>
 
       <p></p>
 
+      <button type="button" class="cd-button cd-button--small cd-button--icon">
+        <svg class="cd-icon cd-icon--user" aria-hidden="true" focusable="false" width="16" height="16">
+          <use xlink:href="#cd-icon--user"></use>
+        </svg>
+        <span class="cd-button__text">Button G</span>
+      </button>
+
+      <p></p>
+
 
       <button type="button" class="cd-button cd-button--export cd-button--icon">
         <span class="cd-button__text">Export</span>
-        <svg class="cd-icon cd-icon--arrow-down">
+        <svg class="cd-icon cd-icon--arrow-down" aria-hidden="true" focusable="false" width="16" height="16">
```

Comment:
Not related to `cd-button` in particular, but I'm starting to wish for some shorthand to output an SVG from our sprite with all these props that get repeated.

```
<deco-svg icon="arrow-down">
```

That seems to be about all that is needed to expand into the full markup with class and `xlink:href` anchor thing.

Reply:
If the idea is to avoid writing the full markup in templates, we could probably extend twig with a function that would output the markup like `{{ cd_svg_icon('arrow-down') }}`.

Reply:
Adding as a backlog item

Reply:
@orakili yes, exactly the type of solution I was thinking of. my sample markup was pseudo-code.

We do have a Vue component to use as a model, which allows additional classes and a custom fill color: https://github.com/UN-OCHA/hid_v3/blob/dev/components/CommonIcon.vue